paddle/gserver/gradientmachines/MultiGradientMachine.cpp

```diff
@@ -601,7 +601,7 @@ void TrainerThread::backward() {
 
 void TrainerThread::backwardCallback(Parameter* para) {
   // CPU parameters are merged in the end
-  if (!para->useGpu()) return;
+  if (!para->useGpu() || para->isStatic()) return;
```

Comment:
If the batch size for one thread is 0, all the parameters will call `backwardCallback ` function to merge gradient in the line [593](https://github.com/PaddlePaddle/Paddle/pull/2650/files#diff-13b947557637acb90d7b810c647bf277R593). But the static parameters which do not need to be updated should not call `backwardCallback `. So add a check here.